PowerEditor/src/WinControls/OpenSaveFileDialog/CustomFileDialog.cpp

```diff
@@ -617,15 +630,88 @@ class FileDialogEventHandler : public IFileDialogEvents, public IFileDialogContr
 
 	static LRESULT CALLBACK KbdProcHook(int nCode, WPARAM wParam, LPARAM lParam)
 	{
+		static bool isCandidateMode = false;
+		static bool wasCandidateMode = false;
+		auto IsHangul = [](WCHAR locale[])
+		{
+			auto hwnd = GetFocus();
+			auto himc = ImmGetContext(hwnd);
+			auto isHangul = ImmGetOpenStatus(himc);
+			ImmReleaseContext(hwnd, himc);
+
+			return !wcscmp(locale, L"ko-KR") && isHangul;
```

Comment:
1. Add a null check for `locale`.
2. Is this condition needed only for Korean? Do we need to check any other language?